### PR TITLE
Feature/blui 5159 registration workflow context

### PR DESCRIPTION
**Pull request by ektaghag-eaton**
Fixes 5159

#### Changes proposed in this Pull Request:
- Demo screens will be removed in future
- Added registrationWorkflowContext
- Added demo screens
- Update Docs

#### Screenshots / Screen Recording (if applicable)

- 

Uploading With_CustomScreen.mp4…

Uploading Default_Workflow.mp4…

#### To Test:

-Select `RegistrationProviderExample` from the drawer menu and observe all data persists on the back and next button

#### Any specific feedback you are looking for?

- **This PR is only to showcase how registration workflow context persists data. Please do not focus on individual screen's UI or functionality.**

### CODE REVIEW

**Comment by ArshdeepSingh3-eaton** [comment]
everything looks good
should it have documentation for registration workflow context ?

**Comment by ektaghag-eaton** [comment]
> everything looks good but should it have documentation for registration workflow context ?

Hi @ArshdeepSingh3-eaton , updated docs. Please review again.

File: login-workflow/src/contexts/RegistrationWorkflowContext/RegistrationWorkflowContext.test.tsx
Comment (surajeaton) [inline login-workflow/src/contexts/RegistrationWorkflowContext/RegistrationWorkflowContext.test.tsx:1]: Please this test to __test__ folder. Also create contexts folder

Comment (surajeaton) [inline login-workflow/src/contexts/RegistrationWorkflowContext/RegistrationWorkflowContext.test.tsx:1]: Also we are following the convention Comp-test.tsx

**Review by ArshdeepSingh3-eaton — APPROVED** [review]

**Pull request merged**